src/worker/main.ts

```diff
@@ -1 +1,66 @@
-self.postMessage("hello, from worker");
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
+
+/**
+ * A "fake" type; it's a string, but its contents MUST be a valid BCP-47 tag.
+ */
+type BCP47 = string & { __bcp46__: true };
+
+/**
+ * A model ID, according to spec.
+ */
+interface ModelID {
+  readonly author: string;
+  readonly language: string;
+  readonly tag: string;
+}
+
+interface CompileMessage {
+  readonly method: "compile";
+  readonly modelID: ModelID;
+  readonly files: { [filename: string]: ArrayBuffer };
+}
+
+self.onmessage = (ev: MessageEvent) => {
+  if (isCompileMessage(ev.data)) {
+    compileModel(ev.data)
+      .then((code) => {
+        self.postMessage(code);
+      })
+      .catch((e) => {
+        self.postMessage({ error: e });
+      });
+  }
+};
+
+async function compileModel(spec: CompileMessage): Promise<string> {
+  const v = Object.entries(spec.files)[0];
+  const source = await parseFile(v[0], v[1]);
```

Comment:
**[Question]**: if `await`is used here this function is essentially a synchronous function and `async` can be removed

Reply:
[`await` is only syntactically valid in a function marked as `async`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#:~:text=The%20await%20operator%20is%20used%20to%20wait%20for%20a%20Promise.%20It%20can%20only%20be%20used%20inside%20an%20async%20function.)

Reply:
hahaha nice doc for my rusty async implementation